node/http.go

```diff
@@ -86,6 +87,11 @@ func makeRPCOverHTTP(host string, port uint16, servers map[string]*jsonrpc.Serve
 			httpHandler = httpHandler.WithListener(listener)
 		}
 		mux.Handle(path, exactPathServer(path, httpHandler))
+
+		prefixedPath := strings.TrimSuffix(path, "/")
+
+		mux.HandleFunc(prefixedPath+"/ready", rpchandler.HandleReadyRequest)
+		mux.HandleFunc(prefixedPath+"/ready/sync", rpchandler.HandleReadySyncRequest)
```

Comment:
I think the best way to structure it would be: register `/ready` path in special http readiness handler (see `servers` variable). It would give us these benefits:

1. reduce number of readiness paths
2. reuse same logic for path handling assignments
3. separation of conerns: jsonrpc server struct should handle only corresponding requests and separate structure for readiness checks should handle its own kind of requests

wdyt?

Reply:
This sounds like the correct way. I separated the logic from jsonrpc server and moved it into a special http handler. Reduced the number of paths serving for readiness.